lib/internal/url.js

```diff
@@ -320,6 +320,64 @@ class URL {
           onParseError);
   }
 
+  static from(input) {
+    input = input || {};
+
+    const {
+      username = '',
+      password = '',
+      host = null,
+      port = null,
+      query = null,
+      path = [],
+      fragment = null,
+    } = input;
+    const protocol = `${input.protocol || ''}:`;
+
+    let flags = 0;
+
+    if (
```

Comment:
What if we instead wrote this more like we do in the existing codebase?:

```js
    if (protocol === 'file:' ||
        protocol === 'https:' ||
        protocol === 'wss:' ||
        protocol === 'http:' ||
        protocol === 'ftp:' ||
        protocol === 'ws:' ||
        protocol === 'gopher:') {
      flags |= URL_FLAGS_SPECIAL;
    } else if (!host) {
      flags |= URL_FLAGS_CANNOT_BE_BASE;
    }
```

Reply:
What do you mean? Where would be such example?

Reply:
Example of what it would look like? I provided one above.